examples/batch/batch.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	pgx "github.com/jackc/pgx/v5"
+	pgconn "github.com/jackc/pgx/v5/pgconn"
+	pgxpool "github.com/jackc/pgx/v5/pgxpool"
+)
+
+type PgxPoolInterface interface {
+	Begin(context.Context) (pgx.Tx, error)
+	Exec(context.Context, string, ...interface{}) (pgconn.CommandTag, error)
+	SendBatch(context.Context, *pgx.Batch) pgx.BatchResults
+	Close()
+}
+
+type ExampleBatch struct {
+	batch  *pgx.Batch
+	br     pgx.BatchResults
+	db     PgxPoolInterface
+	output strings.Builder
+}
+
+func (ex *ExampleBatch) insertRow(fields ...string) (err error) {
```

Comment:
I don't see any value in `insertRow` function. And it's not covered by tests. It's better to remove it and keep things simple